types/components/buttons.d.ts

```diff
@@ -108,6 +108,81 @@ export type OnShippingChangeActions = {
     };
 };
 
+type CurrencyCodeAndValue = {
+    currencyCode: string;
+    value: string;
+};
+
+type ShippingOption = {
+    amount: CurrencyCodeAndValue;
+    id?: string;
+    label: string;
+    selected: boolean;
+    type: "SHIPPING" | "PICKUP";
+};
+
+export type OnShippingOptionsChangeData = {
+    orderID?: string;
+    paymentID?: string;
+    paymentToken?: string;
+    selectedShippingOption?: ShippingOption;
+};
+
+type BuildOrderPatchPayloadOptions = {
+    discount?: string;
+    handling?: string;
+    insurance?: string;
+    itemTotal?: string;
+    option?: ShippingOption;
+    shippingOptions?: ShippingOption[];
+    shippingDiscount?: string;
+    taxTotal?: string;
+};
```

Comment:
How about using `BuildOrderPatchPayloadArgs` [like in SPB](https://github.com/paypal/paypal-smart-payment-buttons/blob/main/src/props/onShippingAddressChange.js#L34-L42)?